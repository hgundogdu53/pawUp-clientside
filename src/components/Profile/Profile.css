h4.profile-title {
    font-size: 1.75rem;
    animation-name: moveInLeft;
    animation-duration: 2s;
    animation-timing-function: ease-in;
}

.profile-section {
    transition: 0.3s;
    border-radius: 5px;
    width: 22.5rem;
    margin-top: 0.5rem;
    background-color: rgba(202,244,241, 0.2);
    width: 30rem;
}

.update-post-section:hover {
    box-shadow: 0 8px 16px 0 rgba(255, 255, 255, 0.151);
}