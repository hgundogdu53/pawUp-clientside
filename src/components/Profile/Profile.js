import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import config from '../../config';
import TokenService from '../../services/token-service';
import './Profile.css';


class Profile extends Component {
    constructor(props) {
        super(props);
        this.state = {
            users: []
        }
    }

    componentDidMount() {
        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `bearer ${TokenService.getAuthToken()}`
            }
        }

        fetch(`${config.API_BASE_URL}/users/user`, options)
            .then(res => {
                if (res.ok) {
                    return res.json();
                }
                else {
                    throw new Error('Something went wrong');
                }
            })
            .then(data => {
                this.setState({ users: data })
            })
            .catch(err => {
                throw new Error(err)
            }
            )
    }

    render() {
        return (
            <main className='flex-container'>
                <header role="banner">
                    <h2>My Profile</h2>
                </header>
                <section key={this.state.users.id} className='profile-section'>
                    <h4 className='profile-title'>{this.state.users.first_name} {this.state.users.last_name}</h4>
                    <p>Email: {this.state.users.email}</p>
                    <button className='deletePostBtn' onClick={() => {
                        this.props.deleteUser(this.state.users.id, this.deleteUser)
                    }}>Delete</button>
                    <Link to='/editProfile'><button className='update-btn'>Update</button></Link>
                </section>
            </main>
        )
    }
}

export default Profile;