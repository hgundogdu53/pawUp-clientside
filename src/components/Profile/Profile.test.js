import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Profile from './Profile';

describe(`Profile component`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<Profile />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});