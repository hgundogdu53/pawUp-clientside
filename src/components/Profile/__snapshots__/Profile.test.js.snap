// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders without crashing 1`] = `
<main
  className="flex-container"
>
  <header
    role="banner"
  >
    <h2>
      My Profile
    </h2>
  </header>
  <section
    className="profile-section"
  >
    <h4
      className="profile-title"
    >
       
    </h4>
    <p>
      Email: 
    </p>
    <button
      className="deletePostBtn"
      onClick={[Function]}
    >
      Delete
    </button>
    <Link
      to="/editProfile"
    >
      <button
        className="update-btn"
      >
        Update
      </button>
    </Link>
  </section>
</main>
`;
